src/components/Offer/index.tsx

```diff
@@ -1,31 +1,114 @@
-import React from 'react';
+import React, { useState }  from 'react';
+import { useConnection, useWallet } from '@solana/wallet-adapter-react'
 import { useForm } from 'react-hook-form';
 import { Link } from 'react-router-dom';
+import { AuctionHouseProgram } from '@metaplex-foundation/mpl-auction-house';
+import { MetadataProgram } from  '@metaplex-foundation/mpl-token-metadata';
+import { Transaction, PublicKey } from '@solana/web3.js';
+import BN from 'bn.js';
 
+const NATIVE_MINT = new PublicKey("So11111111111111111111111111111111111111112")
 interface OfferForm {
   amount: string;
 }
 
 interface OfferProps {
   nft: any;
+  marketplace: any;
```

Comment:
```suggestion
  nft: Nft;
  marketplace: Marketplace;
```

can get from the types file.

Reply:
great point